src/library/Truncate/Truncate.js

```diff
@@ -0,0 +1,90 @@
+/* @flow */
+import React, { Component } from 'react';
+import { canUseDOM } from 'exenv';
+import debounce from 'lodash.debounce';
+import { ellipsis } from 'polished';
+import { createStyledComponent } from '../styles';
+import _Tooltip from '../Tooltip';
+
+type Props = {
+  /** Content of Truncate */
+  children?: React$Node
+};
+
+type State = {
+  tooltip: boolean
+};
+
+const styles = {
+  root: ({ theme: baseTheme }) => ({
+    pointerEvents: 'all', // Necessary because of Button Inner's pointerEvents: none
+    ...ellipsis('100%'),
+
+    '&:focus': {
+      outline: `1px solid ${baseTheme.color_theme}`,
+      outlineOffset: -1
+    }
+  }),
+  tooltip: {
+    // This style is only necessary because we're using Truncate as a Button child
+    whiteSpace: 'normal'
+  }
+};
+
+const Root = createStyledComponent('span', styles.root, {
+  displayName: 'Truncate'
+});
+const Tooltip = createStyledComponent(_Tooltip, styles.tooltip);
+
+export default class Truncate extends Component<Props, State> {
+  state = {
+    tooltip: false
+  };
+
+  root: HTMLElement;
+
+  setRootRef = (node: HTMLElement) => {
+    this.root = node;
+  };
+
+  addTooltip = () => {
```

Comment:
Is addTooltip the best name for this method? Won't it also remove it if it is no longer necessary?

Reply:
Changed to `toggleTooltip`.